Modules/Core/Common/include/itkImageToImageFilter.hxx

```diff
@@ -184,10 +184,7 @@ ImageToImageFilter<TInputImage, TOutputImage>::VerifyInputInformation() ITKv5_CO
       const SpacePrecisionType coordinateTol =
         itk::Math::abs(this->m_CoordinateTolerance * inputPtr1->GetSpacing()[0]); // use first dimension spacing
 
-      if (!inputPtr1->GetOrigin().GetVnlVector().is_equal(inputPtrN->GetOrigin().GetVnlVector(), coordinateTol) ||
-          !inputPtr1->GetSpacing().GetVnlVector().is_equal(inputPtrN->GetSpacing().GetVnlVector(), coordinateTol) ||
-          !inputPtr1->GetDirection().GetVnlMatrix().as_ref().is_equal(inputPtrN->GetDirection().GetVnlMatrix().as_ref(),
-                                                                      this->m_DirectionTolerance))
+      if (!inputPtr1->SamePhysicalSpaceAs(inputPtrN, m_CoordinateTolerance, m_DirectionTolerance))
       {
         std::ostringstream originString, spacingString, directionString;
         if (!inputPtr1->GetOrigin().GetVnlVector().is_equal(inputPtrN->GetOrigin().GetVnlVector(), coordinateTol))
```

Comment:
Maybe the redundancy of duplicate `is_equal` calls that was in the original code already should also be addressed. I mean, it now calls `is_equal` twice for each of the three properties, Origin, Spacing, and Direction. I think it would be better to store the intermediate `is_equal` results, to avoid duplicated `is_equal` calls: 

```cpp
      const bool equalOrigin = inputPtr1->GetOrigin().GetVnlVector().is_equal(...);
      const bool equalSpacing = inputPtr1->GetSpacing().GetVnlVector().is_equal(...);
      const bool equalDirection = inputPtr1->GetDirection().GetVnlMatrix().as_ref().is_equal(...);

      if (!equalOrigin || !equalSpacing || !equalDirection)
      {
        std::ostringstream originString, spacingString, directionString;
        if (!equalOrigin) {...}
        if (!equalSpacing) {...}
        if (!equalDirection) {...}
```

Reply:
I am more concerned about keeping the duplicated `is_equal` calls in sync. Each of the three conditions in `if (!equalOrigin || !equalSpacing || !equalDirection)` should exactly match the individual checks below there, `if (!equalOrigin)`, `if (!equalSpacing)`, and `if (!equalDirection)`.